lib/CL/pocl_llvm_spirv.cc

```diff
@@ -283,31 +294,36 @@ int pocl_regen_spirv_binary(cl_program program, cl_uint device_i) {
   int spec_constants_changed = 0;
   char concated_spec_const_option[MAX_SPEC_CONST_CMDLINE_LEN] = {};
   concated_spec_const_option[0] = 0;
-  char spirv_exts[MAX_SPEC_CONST_CMDLINE_LEN] = {};
+  std::string SpirvExts;
+  std::string SpirvMaxVersion;
   char program_bc_spirv[POCL_MAX_PATHNAME_LENGTH] = {};
   char unlinked_program_bc_temp[POCL_MAX_PATHNAME_LENGTH] = {};
   program_bc_spirv[0] = 0;
   unlinked_program_bc_temp[0] = 0;
 
+  pocl_version_t MaxV;
+  getMaxSpirvVersion(MaxV, Device->supported_spir_v_versions);
+  SpirvMaxVersion = "--spirv-max-version=" + std::to_string(MaxV.major) +
+                    "." + std::to_string(MaxV.minor);
```

Comment:
FWIW, this is not needed if the module's target triple has version suffix (e.g. `spirv64v1.3`).

Reply:
This is to @franz also

Reply:
It seems that at least in some cases, it doesn't have the version suffix, because i was getting the "Invalid SPIR-V module: incorrect SPIR-V version number" from the reverse translation code (with ocloc).